### ISSUE TITLE
Error during post cluster up configuration: Error reading config file ~/.minishift/machines/minishift_kubeconfig

### ISSUE THREAD

**Issue by indrayam** [minishift/minishift#1969]
### General information

  * Minishift version: v1.12.0+daa0943
  * OS: macOS
  * Hypervisor: VirtualBox

I am running 5.2.6 r120293 of VirtualBox. I already have minikube, gcloud (with kubectl) and Docker for Mac (Edge Channel) with Kubernetes support installed. I connect to k8s clusters hosted at GKE, locally on minikube and by Docker for Mac. Which is to say, I have an existing ~/.kube/config file.

I wanted to install minishift to try out the Istio Tutorial hosted by RedHat [here](https://github.com/redhat-developer-demos/istio-tutorial)

### Steps to reproduce

  1. Download and installed minishift in `/usr/local/bin`
  2. Run `(prompt)>minishift start --vm-driver=virtualbox`
  3. Things seem to work well until I get the following error:
```
Error during post cluster up configuration: Error reading config file /Users/anasharm/.minishift/machines/minishift_kubeconfig
```

### Expected
I was hoping that the post cluster configuration would work. It seems like I can run most of the commands (like `minishift status` and `oc get projects`) and they all seem to work right, but I saw this open issue - [#1827](https://github.com/minishift/minishift/issues/1827). Sure enough, I ran into the same issue as well...

```
(prompt)>minishift openshift service nodejs-ex --in-browser
Error getting valid namespaces user belongs to.: exit status 1
```

### Actual
Since there is no `minishift_kubeconfig` file, guess the error was expected. The `oc` command did not work. I found the binary inside `~/.minishift/cache/oc` folder and had to create a symlink in `/usr/local/bin`. 

### Logs
Output from `minishift start --show-libmachine-logs -v5` command is at this location: [https://gist.github.com/indrayam/51f31abbbd67314a083e98d846ccc9e0](https://gist.github.com/indrayam/51f31abbbd67314a083e98d846ccc9e0)

**Comment by praveenkumar** [comment]
@indrayam This might happen because the way we parse the ~/.kube/config and extract minishift specific details but in your case looks like we are not parsing it well enough and `minishift_kubeconfig` file become corrupted. Can you please provide how is your ~/.kube/config file looks like (make sure you remove all the private token from it before pasting it)?

**Comment by indrayam** [comment]
Here you go @praveenkumar 
[config.txt](https://github.com/minishift/minishift/files/1691158/config.txt)

I have tried to clean up all my secrets by adding a "..." in between start and end characters. Hope I did not miss any :wink:

**Comment by praveenkumar** [comment]
Thanks @indrayam, will look into this and let you know if any more information required from your end.

**Comment by Raboo** [comment]
I have the same issue, I suspect because i've been playing with GKE and it adds stuff to ~/.kube/config

**Comment by indrayam** [comment]
@Raboo Agreed. I had to move `~/.kube` to `~/.kube.orig` before running `minishift start` command. I talk about it more here:
https://github.com/minishift/minishift/issues/2026#issuecomment-365756697

**Comment by budhrg** [comment]
> Agreed. I had to move ~/.kube to ~/.kube.orig before running minishift start command. I talk about it more here:

Yes, it is related to `~/.kube/config`. We are parsing it and creating `minishift_kubeconfig`. It might happen that the parsing is not happening happily here.

We are investigating it and coming up with better/alternate solution.

**Comment by praveenkumar** [comment]
@indrayam Can you please try the artifact from #2034 PR which is available https://ci.appveyor.com/project/budhrg/minishift/build/275/artifacts and then let me know if you still need to backup your config before using minishift.

**Comment by jorgemoralespou** [comment]
@budhrg why do you need to parse copy the .kube/config file?
Can't you use a KUBECONFIG env to use a new/fresh .kube/config in the specified location?

**Comment by indrayam** [comment]
@praveenkumar Do you want me to download the minishift at the URL you shared and see if it works? I am sorry. Did not check my notifications over the weekend

**Comment by praveenkumar** [comment]
> Do you want me to download the minishift at the URL you shared and see if it works? 

Yes, that what I wanted to make sure.

> I am sorry. Did not check my notifications over the weekend

no problem :)

**Comment by Raboo** [comment]
@praveenkumar that artifact works without a problem!

**Comment by indrayam** [comment]
@praveenkumar Worked like a charm. Now, I have another pet peeve. Before installing minishift, I had 5 contexts:
```
az-anand-cluster
do-anand-cluster
do-anand-cluster-self
gke-anand-cluster
os-anand-cluster
```

After installing Minishift, I ended up with 11!
```
az-anand-cluster
do-anand-cluster
do-anand-cluster-self
gke-anand-cluster
os-anand-cluster
minishift
myproject/192-168-99-100:8443/developer
myproject/192-168-99-100:8443/system:admin
default/127-0-0-1:8443/system:admin
default/192-168-99-100:8443/system:admin
/192-168-99-100:8443/developer
```

I do not know enough about OpenShift at this stage to know what they all do. **Do I really need them all?** 

It seems like the `minishift start` command could offer some options to opt-out of the ones that may not be necessary. Or better yet, create simpler context names :wink:

**Comment by jorgemoralespou** [comment]
@indrayam See https://github.com/openshift/origin/issues/18645

**Comment by indrayam** [comment]
Awesome. Thanks for sharing @jorgemoralespou

**Comment by praveenkumar** [comment]
> Worked like a charm. 

@indrayam Great, thanks for testing.

**Comment by LalatenduMohanty** [comment]
Resolved by #2034
